mercury/v1/mercury.go

```diff
@@ -0,0 +1,465 @@
+package mercury_v1
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"sort"
+	"time"
+
+	pkgerrors "github.com/pkg/errors"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+
+	"github.com/smartcontractkit/chainlink-mercury/mercury"
+	"github.com/smartcontractkit/chainlink-relay/pkg/logger"
+	relaytypes "github.com/smartcontractkit/chainlink-relay/pkg/types"
+)
+
+// MaxAllowedBlocks indicates the maximum len of LatestBlocks in any given observation.
+// observations that violate this will be discarded
+const MaxAllowedBlocks = 5
+
+// Mercury-specific reporting plugin, based off of median:
+// https://github.com/smartcontractkit/offchain-reporting/blob/master/lib/offchainreporting2/reportingplugin/median/median.go
+
+type Observation struct {
```

Comment:
Hmmm these are used in core telemetry so it seems like they should be left in -relay... :thinking: